src/librados/AioCompletionImpl.h

```diff
@@ -85,10 +85,7 @@ struct librados::AioCompletionImpl {
     return wait_for_complete();
   }
   int is_complete() {
-    lock.Lock();
-    int r = complete;
-    lock.Unlock();
-    return r;
+    return complete;
```

Comment:
you are repeating https://github.com/ceph/ceph/pull/14026#discussion_r119541510. please feel free to post a PR, if you believe it's a bottle neck.